### PR TITLE
chore: make `--driver-write-secrets` flag no-op

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

Driver write secrets feature has been enabled by default in `v0.1.0` release. This PR makes the flag configuration no-op to not allow disabling the feature. The flag will be removed altogether from code, deployment and helm charts in n+2 releases.

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -71,14 +75,14 @@ func main() {
 	if *provider.ConstructPEMChain {
 		klog.Infof("construct pem chain feature enabled")
 	}
-	if *provider.DriverWriteSecrets {
-		klog.Infof("secrets will be written to filesystem by the CSI driver")
+	if !*driverWriteSecrets {
+		klog.Infof("driver write secrets feature can't be disabled. The --driver-write-secret flag will be removed in future releases.")
```
Comment (nilekhc) [inline cmd/main.go:79]: Should we do structured logging here?

**Comment by nilekhc** [comment]
@aramase just a small nit. Otherwise looks good.

File: cmd/main.go (hunk 1 above)
Comment (aramase) [reply cmd/main.go:79]: There are no key-values in this log statement, so this is also structure logged.

**Review by nilekhc — APPROVED** [review]
lgtm

**Pull request merged**